app/code/Magento/Catalog/view/adminhtml/templates/catalog/product/attribute/set/main.phtml

```diff
@@ -159,7 +159,7 @@ script;
                             for( i in rootNode.childNodes ) {
                                 if(rootNode.childNodes[i].id) {
                                     var group = rootNode.childNodes[i];
-                                    editSet.req.groups[gIterator] = new Array(group.id, group.attributes.text.strip(),
+                                    editSet.req.groups[gIterator] = new Array(group.id, Ext.util.Format.htmlDecode(group.attributes.text.strip()),
```

Comment:
We already have JS method for escaping HTML here:
https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/Security/view/base/web/js/escaper.js#L38
Feels like it does the same thing.

Reply:
Actually, it does the opposite.

Also, the reason why I went with adding `Ext.util.Format.htmlDecode`  (as a counterpart to the existing `Ext.util.Format.htmlEncode`) in [ext-tree.js](https://github.com/magento/magento2/blob/2.4-develop/lib/web/extjs/ext-tree.js) is because it lives outside of the module system.

Reply:
So, feels like we shouldn't escape it here: https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/Catalog/Block/Adminhtml/Product/Attribute/Set/Main.php#L236

And add escaping on the JS side.
The main idea there - prevent potential XSS.

Reply:
The decoded value here is only used in the save request (this is a change to [`rebuildTrees`](https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/Catalog/view/adminhtml/templates/catalog/product/attribute/set/main.phtml#L155-L193), that is only used to serialize the tree when the attribute set is [saved](https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/Catalog/view/adminhtml/templates/catalog/product/attribute/set/main.phtml#L389)).

Reply:
Ok, so I'm removing myself from assignee, probably someone else will review it deeper